feat(transformer/statement-injector): add an assertion to check if it still contains statements that don't inject

Dunqing opened the pull request:
When there are still statements that aren't inserted after the visitor ends, it means statements inserted with an incorrect address or outdated address (e.g., the original statement has mutated to a new statement). 

This PR adds an assertion to expose this potential bug as soon as possible on the development side.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9063?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9063** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9063?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9063?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9062** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9062?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9061** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9061?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9063).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented on crates/oxc_transformer/src/common/statement_injector.rs:
```diff
@@ -250,4 +255,12 @@ impl<'a> StatementInjectorStore<'a> {
+    // `#[inline(always)]` because this is a no-op in release mode
+    #[expect(clippy::inline_always)]
+    #[inline(always)]
+    fn assert_insertions(&self) {
```
I suggest giving this method a more descriptive name e.g. `assert_no_insertions_remaining`.

Dunqing replied:
Looks good to me!

overlookmotel reviewed (approved):
This is a really good idea. You can ignore my nit about naming if you can't be bothered!

overlookmotel commented:
### Merge activity

* **Feb 18, 4:48 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
feat(transformer/statement-injector): add an assertion to check if it still contains statements that don't inject (#9063)

When there are still statements that aren't inserted after the visitor ends, it means statements inserted with an incorrect address or outdated address (e.g., the original statement has mutated to a new statement).

This PR adds an assertion to expose this potential bug as soon as possible on the development side.
